src/library_pthread.js

```diff
@@ -1037,15 +1037,11 @@ var LibraryPThread = {
     return {{{ makeDynCall('ii', 'ptr') }}}(arg);
   },
 
-  // This function is called internally to notify target thread ID that it has messages it needs to
-  // process in its message queue inside the Wasm heap. As a helper, the caller must also pass the
-  // ID of the main browser thread to this function, to avoid needlessly ping-ponging between JS and
-  // Wasm boundaries.
-  _emscripten_notify_thread_queue: function(targetThreadId, mainThreadId) {
-    if (targetThreadId == mainThreadId) {
-      postMessage({'cmd' : 'processQueuedMainThreadWork'});
+  _emscripten_notify_proxying_queue: function(targetThreadId, currThreadId, mainThreadId, queue) {
+    if (targetThreadId == currThreadId) {
+      setTimeout(function() { _emscripten_proxy_execute_queue(queue); });
```

Comment:
Why turn this into a setTimeout?

Reply:
Actually, this new `_emscdripten_notify_proxying_queue` is a duplicate of the one that already exists just below. I'm not sure how I ended up with the duplicate function, but I'll remove it. The diff here should simply be a removal of the old `_emscripten_notify_thread_queue`.

Reply:
The reason this is `setTimeout` is so that we process the queue after returning to the event loop, similar to how queues on other threads will only be processed once they return to their event loops.

Reply:
Interesting.. I didn't see that the condition here changed from `(targetThreadId == mainThreadId)` to `(targetThreadId == currThreadId)`.

How was the `(targetThreadId == currThreadId)` handled in the old code.. it looks like it was not handled specially and

Reply:
The old code eagerly executed the proxied work if the target thread was the same as the current thread, so `_emscripten_notify_thread_queue` was never called in that case. The new API is more flexible and does not force eager evaluation in that case, so the new JS code has to handle it.